### ISSUE TITLE
Windows cannot access the folder where the download happens during 'upgrade'

### ISSUE THREAD

**Issue by coeus89** [microsoft/winget-cli#2984]
### Brief description of your issue

when i try a winget upgrade 'package' for a package that installs in program files, i get the attached error. I am running in standard user mode. I am sure that admin has access to the folder.
![image](https://user-images.githubusercontent.com/18616316/219882585-41ed4f3a-2d27-4311-b58e-92a35fdeafe6.png)

### Steps to reproduce

type winget upgrade 'package name'
wait for the download
type in the admin password
get the error message

### Expected behavior

I expected the program to install properly

### Actual behavior

The application errors out and does not install if it requires admin

### Environment

```shell
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1265
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by coeus89** [comment]
This happens for most applications that require admin. something about not being able to access the network or directory. but the directory has full privileges.

**Comment by coeus89** [comment]
![image](https://user-images.githubusercontent.com/18616316/219883296-c1ba88ea-5745-4648-891e-f2e448fe54a5.png)

**Comment by PapysGit** [comment]
> ### Brief description of your issue
> when i try a winget upgrade 'package' for a package that installs in program files, i get the attached error. I am running in standard user mode. I am sure that admin has access to the folder. ![image](https://user-images.githubusercontent.com/18616316/219882585-41ed4f3a-2d27-4311-b58e-92a35fdeafe6.png)
> 
> ### Steps to reproduce
> type winget upgrade 'package name' wait for the download type in the admin password get the error message
> 
> ### Expected behavior
> I expected the program to install properly
> 
> ### Actual behavior
> The application errors out and does not install if it requires admin
> 
> ### Environment
> ```shell
> Windows Package Manager v1.4.10173
> Copyright (c) Microsoft Corporation. All rights reserved.
> 
> Windows: Windows.Desktop v10.0.22621.1265
> System Architecture: X64
> Package: Microsoft.DesktopAppInstaller v1.19.10173.0
> 
> Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
> 
> User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
> 
> Links
> ---------------------------------------------------------------------------
> Privacy Statement   https://aka.ms/winget-privacy
> License Agreement   https://aka.ms/winget-license
> Third Party Notices https://aka.ms/winget-3rdPartyNotice
> Homepage            https://aka.ms/winget
> Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
> ```

Facing the exact same issue.

**Comment by jdelker** [comment]
The problem is with the temp directory winget uses for install.
This is located at `%LOCALAPPDATA%\temp\WinGet` and simply has insufficient permissions for an elevated permission install.
So effectively the given admin account is unable to read files from this directory.
Unfortunately, it doesn't help to adjust the permissions for this directory manually, because winget resets it during upgrade.

This problem already persists for months now and I've no clue why this hasn't been fixed yet. Actually it should affect many users, if not all uses.

**Comment by jmeile** [comment]
I saw this too. When trying to install "7zip.7zip":
`Winget install 7zip.7zip`

I got the problem mentioned by the original poster. It seems to be related to the repository and happening with user elevation. If I use this instead:
`winget` install "7-zip" --source msstore

Then it will install. So, I guess other repositories are buggy

**Comment by Prometheus720** [comment]
This has been a consistent problem for me and a main reason why I must still rely on other package managers.

**Comment by loooooooooooor** [comment]
I face the exact same issue.

**Comment by static0verdrive** [comment]
I also face this issue for anything requiring admin. I was trying to install Oracle.Virtualbox but I also tried to upgrade existing packages, and it failed with the same network error.

**Comment by niStee** [comment]
> The problem is with the temp directory winget uses for install. This is located at `%LOCALAPPDATA%\temp\WinGet` and simply has insufficient permissions for an elevated permission install. So effectively the given admin account is unable to read files from this directory. Unfortunately, it doesn't help to adjust the permissions for this directory manually, because winget resets it during upgrade.
> 
> This problem already persists for months now and I've no clue why this hasn't been fixed yet. Actually it should affect many users, if not all uses.

What I can't understand is why it doesn't affect every package. Some of the programs I have installed can still be updated via winget, others can't.

**Comment by io43** [comment]
I just found some workaround, the issue isn't that much widespread because it seem to happen only when you use a simple user as account instead of the default account administrator type, so as workaround you can launch the console as your other admin  user with runas  and run winget from there

**Comment by jmeile** [comment]
@io43
I tried to do what you said, But from the admin command, I get this:
```
Winget uninstall 7zip.7zip
No installed package found matching input criteria.
```

It only works from my normal user. I tried with the "Runas" command with different switches: "/profile", "/noprofile", "/env", and no switch at all and I always get that error. I even tried running "cmd" from the start menu as administrator, but still the same result

**Comment by thesobercoder** [comment]
I moved away from `winget` to `scoop` because of this specific bug. My work laptop is configured to use a non-admin account for everyday work and occasionally use a different admin account to run stuff as admin.

**Comment by ondrejsv** [comment]
This bug is already fixed in the 1.5 version. 
You may download the preview version in the Releases tab and I can confirm that this issue is no longer present.

**Comment by akaegi** [comment]
I can confirm it's working with the preview version (v1.5.1081-preview) 👍

**Comment by adrian-cancio** [comment]
Yes, wonking wll on lastest preview

**Comment by adrian-cancio** [comment]
Yes, working well on current preview

**Comment by mdanish-kh** [comment]
cc @denelon Resolved in #2945
